src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -194,11 +189,14 @@ private void initialize(Context appContext, String authority, ITokenCacheStore t
             throw new UnsupportedOperationException("Local cache is not supported for broker usage");
         }
         mContext = appContext;
-        mConnectionService = new DefaultConnectionService(mContext);
         checkInternetPermission();
         mAuthority = extractAuthority(authority);
         mValidateAuthority = validateAuthority;
         mTokenCacheStore = tokenCacheStore;
+        
+        if (mTokenCacheStore != null) {
+            mTokenCacheAccessor = new TokenCacheAccessor(mTokenCacheStore, mAuthority);
```

Comment:
better to create token accessor even for null token store and check if store is null inside